src/actor.rs

```diff
@@ -351,6 +365,7 @@ impl<V: ReplicaValidator> Actor<V> {
             ActorEvent::TransferRegistrationSent(e) => {
                 self.account.append(e.debit_proof.signed_transfer.transfer);
                 self.accumulating_validations.clear();
+                self.next_debit_version = self.next_debit_version + 1;
```

Comment:
Hm, I don't think this is right.
What made you put this here?

Reply:
We were always expecting `0` as this value was never updated as far as I could see.

Reply:
It is updated on line 338:
self.next_debit_version = e.id().counter;

It is bumped when a transfer is initiated. This is what prevents more than one transfer to be initiated before it has been completed.

Reply:
Line 241, there's where it should go:

`self.next_debit_version = e.id().counter + 1;`

It's just missing a + 1

This is what prevents initiating a new transfer before the previous is completed.
(See validations in `fn validate()`:

```
 // ensures one debit is completed at a time
        if self.next_debit_version != self.account.next_debit() {
            return Err(Error::from("Current pending debit has not been completed"));
        }
```

`self.account.next_debit()` won't be bumped until the transfer is registered.